packages/api/src/modules/providers/scdl/scdl.grant.parser.ts

```diff
@@ -184,4 +186,32 @@ export default class ScdlGrantParser {
         }
         return { entity, annotations, errors };
     }
+
+    /**
+     * Verifies if any column headers are missing in the file and returns a list of the missing ones.
+     * For each missing header, the developer should check if it is a naming issue,
+     * and if so, add it to the SCDL mapper to prevent the data from being excluded from processing.
+     * @param pathObject
+     * @param data
+     *
+     * @returns void
+     */
+    static verifyMissingHeaders<TypeIn extends BeforeAdaptation>(
+        pathObject: DefaultObject<ParserPath | ParserInfo<TypeIn>>,
+        data: NestedDefaultObject<TypeIn>,
+    ): void {
+        const missingKeys = Object.entries(pathObject)
+            .filter(([key, path]) => {
+                const flatMapper = (Array.isArray(path) ? path : path.path).flat();
+                return !flatMapper.some(lib => lib in data);
```

Comment:
Vu qu'il peut y avoir des valeurs "nestées" qui sont gérées par le `GenericParser` mais pas ici, je crains qu'on ait des faux négatifs

Reply:
Ça arrive quand il y a des sur-catégories : deux lignes de headers dont la première a des cellules fusionnées. Je suis pas sûre qu'on ait croisé le cas pour le scdl, mais pour chorus et osiris je suis presque sûre que ça arrive, donc le `GenericParser` le prend en compte et je trouve dommage/risqué de le zapper ici. Ou alors il faut un commentaire très très voyant pour ne pas créer de confusion

Reply:
Ce que je propose c'est déjà de rajouter un if (DEV) en tête de toute la methode verifyMissingHeaders car ce traitement il est dédié au devs. Ensuite je peux ajouter dans la doc de la methode quelque chose comme "only for SCDL where nested columns are not possible". Je peux aussi passer la fonction private pour éviter quelle soit utilisable en dehors. Qu'en penses tu ?